src/hotspot/share/opto/memnode.cpp

```diff
@@ -3424,7 +3435,7 @@ Node *MemBarNode::Ideal(PhaseGVN *phase, bool can_reshape) {
           eliminate = true;
         }
       }
-    } else if (opc == Op_MemBarRelease) {
+    } else if (opc == Op_MemBarRelease || opc == Op_MemBarStoreStore) {
```

Comment:
Should be protected by a feature flag?

Reply:
I don't think the feature flag is needed for these cases. These are optimizations to support the changes in code shape introduced by the feature, not the feature itself. If the feature flag is off, there should be no change in behavior.

Reply:
Extra safety, though. We are fiddling with common path in the aggressively optimizing compiler. I think we want to have the "chicken flag" that rolls back any changes from this PR, and it is provably the same code as before. For this hunk, it would be something like:

```
} else if (opc == (UseStoreStoreForCtor ? Op_MemBarStoreStore : Op_MemBarRelease)) {
```